src/SFA.DAS.ApplyService.InternalApi.UnitTests/RoatpGatewayOrganisationChecksControllerTests.cs

```diff
@@ -0,0 +1,109 @@
+            _logger = new Mock<ILogger<RoatpGatewayOrganisationChecksController>>();
+            _controller = new RoatpGatewayOrganisationChecksController(_qnaApiClient.Object, _logger.Object);
+        }
+
+        [Test]
+        public void OrganisationChecks_proper_address_is_returned()
+        {
+            var applicationId = Guid.NewGuid();
+            var preambleSequenceNo = 0;
+            var preambleFirstSectionNo = 1;
+
+            var expectedContactAddress = new ContactAddress
+            {
+                Address1 = "First Address",
+                Address2 = "Second Address",
+                Address3 = "Third Address",
+                Address4 = "Fourt Address",
+                Town = "Coventry",
+                PostCode = "CV1 2WT"
+            };
+
+            var returnedPage = new Page { PageOfAnswers = new List<PageOfAnswers> { new PageOfAnswers { Answers = new List<Answer> { 
+                                        new Answer { QuestionId = RoatpPreambleQuestionIdConstants.UkrlpLegalAddressLine1, Value = "First Address"},
+                                        new Answer { QuestionId = RoatpPreambleQuestionIdConstants.UkrlpLegalAddressLine2, Value = "Second Address" },
+                                        new Answer { QuestionId = RoatpPreambleQuestionIdConstants.UkrlpLegalAddressLine3, Value = "Third Address" },
+                                        new Answer { QuestionId = RoatpPreambleQuestionIdConstants.UkrlpLegalAddressLine4, Value = "Fourt Address" },
+                                        new Answer { QuestionId = RoatpPreambleQuestionIdConstants.UkrlpLegalAddressTown, Value = "Coventry" },
+                                        new Answer { QuestionId = RoatpPreambleQuestionIdConstants.UkrlpLegalAddressPostcode, Value = "CV1 2WT" }
+            } } } };
+
+            _qnaApiClient.Setup(x => x.GetPageBySectionNo(applicationId, preambleSequenceNo, preambleFirstSectionNo, RoatpWorkflowPageIds.Preamble)).ReturnsAsync(returnedPage);
+
+            var responseGetOrganisationAddress = _controller.GetOrganisationAddress(applicationId).GetAwaiter().GetResult().Result;
+            var response = responseGetOrganisationAddress as OkObjectResult;
+            var responseContactAddress = response.Value as ContactAddress;
+
+            Assert.AreEqual(expectedContactAddress.Address1, responseContactAddress.Address1);
+            Assert.AreEqual(expectedContactAddress.Address2, responseContactAddress.Address2);
+            Assert.AreEqual(expectedContactAddress.Address3, responseContactAddress.Address3);
+            Assert.AreEqual(expectedContactAddress.Address4, responseContactAddress.Address4);
+            Assert.AreEqual(expectedContactAddress.Town, responseContactAddress.Town);
+            Assert.AreEqual(expectedContactAddress.PostCode, responseContactAddress.PostCode);
+        }
+
+        [Test]
+        public void OrganisationChecks_answer_is_returned()
```

Comment:
Could be more specific about the answer being returned here in the test name

Reply:
I have changed to OrganisationChecks_ico_number_is_returned()